File: 10-20/PatronComposite/src/EjemploGranito/Composite.java
package EjemploGranito;

import java.util.ArrayList;

public class Composite implements ISueldo{
    private ArrayList<ISueldo> empleados= new ArrayList<ISueldo>();
    @Override
    public double getSueldo() {

        double sumador = 0;
        for (int i = 0; i < empleados.size(); i++) {
            sumador = sumador + empleados.get(i).getSueldo();
        }
            return sumador;
        }

      public void agrega(ISueldo p){
            empleados.add(p);
        }

    }

File: 10-20/CompositePython/composite.py
from ISueldo import ISueldo


class Composite(ISueldo):

    def __init__(self):
        self.empleado = []

    def getSueldo(self):
        sumador = 0
        for i in range(len(self.empleado)):
            sumador = sumador + self.empleado[i].getSueldo()
        return sumador

    def agregar(self, p: ISueldo):
        self.empleado.append(p)
